### PR TITLE
LibWasm: Bytecode interpreter performance pack

### PR DESCRIPTION
**Pull request by dzfrias**
Makes fib(30) 2 times faster.

### CODE REVIEW

File: Userland/Libraries/LibWasm/AbstractMachine/AbstractMachine.h [hunk 1]
```diff
@@ -79,41 +79,54 @@ class Value {
...
+
+    template<typename T>
+    requires(sizeof(T) == sizeof(u32)) explicit Value(T raw_value)
+        : m_value(u128(static_cast<i64>(bit_cast<i32>(raw_value)), 0))
     {
     }
 
     template<typename T>
-    requires(sizeof(T) == sizeof(u64)) explicit Value(ValueType type, T raw_value)
+    requires(sizeof(T) == sizeof(u8) && Signed<T>) explicit Value(T raw_value)
```
Comment (dzfrias) [inline Userland/Libraries/LibWasm/AbstractMachine/AbstractMachine.h:95]: I'm certain that there's better template logic that we can do here...

Comment (alimpfard) [reply Userland/Libraries/LibWasm/AbstractMachine/AbstractMachine.h:95]: Eh, looks alright.

**Comment by alimpfard** [comment]
Not a big fan of the inlining, but sure, let's go with it.

**Comment by nico** [comment]
I think something in here, probably the last commit, broke many dbgln_if() statements. If I add `#define WASM_TRACE_DEBUG 1` to the top of (serenity's version of) ByteCodeInterpreter.cpp with this patched in, I get all kinds of diags:

```
../../Userland/Libraries/LibWasm/AbstractMachine/BytecodeInterpreter.cpp:303:69: error: indirection requires pointer operand ('int' invalid)
  303 |     dbgln_if(WASM_TRACE_DEBUG, "map({}) {} = {}", Operator::name(), *value, result);
      |                                                                     ^~~~~~
../../Userland/Libraries/LibWasm/AbstractMachine/BytecodeInterpreter.cpp:891:16: note: in instantiation of function template specialization 'Wasm::BytecodeInterpreter::unary_operation<int, int, Wasm::Operators::EqualsZero>' requested here
  891 |         return unary_operation<i32, i32, Operators::EqualsZero>(configuration);
...
../../Userland/Libraries/LibWasm/AbstractMachine/BytecodeInterpreter.cpp:283:52: error: member reference base type 'int' is not a structure or union
  283 |     dbgln_if(WASM_TRACE_DEBUG, "{} {} {} = {}", lhs.value(), Operator::name(), rhs.value(), result);
      |                                                 ~~~^~~~~~
../../Userland/Libraries/LibWasm/AbstractMachine/BytecodeInterpreter.cpp:893:16: note: in instantiation of function template specialization 'Wasm::BytecodeInterpreter::binary_numeric_operation<int, int, Wasm::Operators::Equals, int>' requested here
  893 |         return binary_numeric_operation<i32, i32, Operators::Equals>(configuration);
```

The diags look right to me. I suppose ladybird's CI doesn't have a bot that builds with dbgln_if()s enabled.

**Comment by nico** [comment]
Also, adding `#define WASI_DEBUG 1` to WASI/Wasi.cpp produces a build error for the `dbgln_if(WASI_DEBUG...` line in that file as well.

**Pull request merged**